Test/dafny0/AttributeChecks.dfy

```diff
@@ -0,0 +1,79 @@
+// RUN: %dafny /print:"%t.print" /dprint:- /compile:0 /env:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+method If(n: int) returns (k: int)
+{
+  var i := 0;
+  if {:split true} i < n {
+    if {:split 1 + 0} true {}
+  }
+  else if {:split false} i > n {}
+  else {:split true} {}
+}
+
+method IfAlt(n: int) returns (k: int)
+{
+  var i := 0;
+  if {:split true}
+  case {:split 1 + true} i < n => // error: 1 + true is ill-typed
+    if {:split 1 + k} true {}
+  case {:split false} i > n  => {}
+  return 2;
+}
+
+method While(n: int) returns (k: int)
+{
+  var f : int -> int := x => x;
+  var i := 0;
+  while {:split true} {:split true + k} i < n // error: true + k is ill-typed
+    invariant forall u :: f(u) == u + i
+  {
+  }
+  return 2;
+}
+
+method WhileAlt(n: int) returns (k: int)
+{
+  var i := 0;
+  while {:split}
+    invariant forall u :: f(u) == u + i
```

Comment:
The `f` on this line is unresolved. If that's intentional, write an `// error: ...` comment on this line. If not, adjust the example (probably by adding a local variable `f`, like in the `While` method above).